lib/cylc/mp_pool.py

```diff
@@ -177,6 +188,7 @@ def process(self):
             else:
                 proc = self._run_command_init(ctx, callback, callback_args)
                 if proc is not None:
+                    ctx.init_time = time.time()
```

Comment:
I wonder if we should store the time out time here, so line 106 will only have to compare current time with the time out time, instead of have to do an addition before the comparison.